src/antelope/chains/Token.ts

```diff
@@ -0,0 +1,125 @@
+import { ethers } from 'ethers';
+
+// A type to represent the possible EVM token types
+export type EvmTokenType = 'ERC20' | 'ERC721' | 'ERC1155' | 'ERC777' | 'ERC827' | 'ERC1400' | 'ERC223';
+
+// A type to represent the source information for a token
+interface TokenSourceInfo {
+    symbol: string;                // Token symbol
+    contract?: string;             // Token contract account name (for native)
+    address?: string;              // Token contract address (for EVM)
+    chainId: string;               // Chain ID (40 & 41 for Telos EVM) or hash (for native)
+    name: string;                  // Token name (as a title)
+    decimals?: number;             // Token amount of digits after the decimal point (as used in EVM)
+    precision?: number;            // Token amount of digits after the decimal point (as used in native)
+    type?: EvmTokenType;           // Token type (ERC20, ERC721, etc.)
+    logo?: string;                 // Token logo uri (as used in native)
+    logoURI?: string;              // Token logo uri (as used in EVM)
+    metadata?: string;             // Token contract metadata (as used in EVM)
+    isSystem: boolean;             // True if the token is the main system token
+    isNative: boolean;             // True if the token is a Antelope native blockchain token (false for EVM)
+}
+
+// A class to represent a blockchain token
+export class Token {
+    readonly id: string;           // Unique ID for the token <symbol>-<contract>-<chainId>
+    readonly symbol: string;       // Token symbol
+    readonly name: string;         // Token name (as a title)
+    readonly logo?: string;        // Token logo uri
+    readonly contract: string;     // Token contract address (for EVM) or account name (for native)
+    readonly chainId: string;      // Chain ID (40 & 41 for Telos EVM) or hash (for native)
+    readonly decimals: number;     // Token amount of digits after the decimal point (same as precision for native)
+    readonly isSystem: boolean;    // True if the token is the system token
+    readonly isNative: boolean;    // True if the token is a native blockchain token
+    private _price: number;        // Token price in USD
```

Comment:
should this be token price in fiat, not USD? also i think this type should be BigNumber or string... do you agree?

Reply:
i think string honestly